packages/plugins/visitor-plugin-common/src/naming.ts

```diff
@@ -0,0 +1,95 @@
+import { ASTNode } from 'graphql';
+import { resolveExternalModuleAndFn } from 'graphql-codegen-plugin-helpers';
+import { NamingConventionMap, NamingConvention, ConvertFn } from './types';
+import { toPascalCase } from './utils';
+
+function getKind(node: ASTNode | string): keyof NamingConventionMap {
+  if (typeof node === 'string') {
+    return 'typeNames';
+  }
+
+  if (['EnumValueDefinition', 'EnumValue'].includes(node.kind)) {
+    return 'enumValues';
+  }
+
+  return 'typeNames';
+}
+
+function getName(node: ASTNode | string): string | undefined {
```

Comment:
nice, so you are passing the entire node, and then based on the type of the node you are converting the name? 
It's much better than we had before (because we needed to pass names each time).

Reply:
YES! And later on we can add more rules to it and we get name conversion in one place.